src/CaptureFile/CaptureFileHelpers.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureFile/CaptureFileHelpers.h"
+
+#include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+
+#include "CaptureFile/CaptureFile.h"
+
+namespace orbit_capture_file {
+
+ErrorMessageOr<void> WriteUserData(
+    const std::filesystem::path& capture_file_path,
+    const orbit_client_protos::UserDefinedCaptureInfo& user_defined_capture_info) {
+  OUTCOME_TRY(capture_file, CaptureFile::OpenForReadWrite(capture_file_path));
+
+  uint32_t message_size = user_defined_capture_info.ByteSizeLong();
+  const size_t buf_size =
+      message_size + google::protobuf::io::CodedOutputStream::VarintSize32(message_size);
+  auto buf = make_unique_for_overwrite<uint8_t[]>(buf_size);
+  google::protobuf::io::ArrayOutputStream array_output_stream{buf.get(),
+                                                              static_cast<int>(buf_size)};
+  google::protobuf::io::CodedOutputStream coded_output_stream{&array_output_stream};
+  coded_output_stream.WriteVarint32(message_size);
```

Comment:
Do we want to document somewhere that we prepend the message with its size? Maybe around here with a comment?

Reply:
Added a section to FORMAT.md file.